src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -30194,6 +30198,105 @@ public static int TakeOutParam<T>(out T x)
             Assert.True(x1.Type.IsErrorType());
         }
 
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/17377")]
+        public void GlobalCode_InferenceFailure_07()
+        {
+            string source =
+@"
+H.M((var x1, int x2));
+H.M(x1);
+
+class H
+{
+    void M(object a) {}
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe.WithScriptClassName("Script"), parseOptions: TestOptions.Script);
+
+            compilation.GetDeclarationDiagnostics().Verify(
+                // (2,10): error CS7019: Type of 'x1' cannot be inferred since its initializer directly or indirectly refers to the definition.
```

Comment:
>// (2,10): error CS7019: Type of 'x1' cannot be inferred since its initializer directly or indirectly refers to the definition. [](start = 16, length = 127)

This error is very misleading, the statement that declares the variable doesn't have any direct or indirect references to it.